### PR TITLE
whitepaper: add whitepaper explaining guardian key usage

### PR DESCRIPTION
**Pull request by tbjump**

### CODE REVIEW

File: whitepapers/0009_guardian_key.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Detailed Design
+
+Signing of gossip messages:
+1. Prepend the message type prefix to the payload
+2. Compute Keccak256Hash of the payload.
+3. Compute ethcrypto.Sign()
+
+Signing of Observations:
+* v1 VAA: `double-Keccak256(observation)`.
+* v2 (batchVAA): `double-Keccak256(version | Keccak256(hash1 | hash2 | ... | hash_n))`, where `|` stands for concatenation.
```
Comment (kcsongor) [inline whitepapers/0009_guardian_key.md:30]: didn't the final design end up with

```
keccak256(version | keccak256(...)))
```

**Review by kcsongor — DISMISSED** [review]
Great! one small suggestion: the title of the document should be Avoiding message confusion, or something along those lines, as it's quite specific to solving that problem

**Review by evan-gray — CHANGES_REQUESTED** [review]
@hendrikhofstadt thoughts on whether this is something for a whitepaper? feels more like an issue description and something that should be in the comments of p2p.go 🤔 thoughts @tbjump?

**Comment by tbjump** [comment]
> @hendrikhofstadt thoughts on whether this is something for a whitepaper? feels more like an issue description and something that should be in the comments of p2p.go 🤔 thoughts @tbjump?

The scope of this is more than p2p.go because the guardian key is used also in the processor for signing observations and *could* be used outside of the guardiand as well. E.g. yesterday there was discussion about using it for wormchain ceremony. I think given the importance and potential for errors, a central whitepaper is appropriate.

**Review by claudijd — DISMISSED** [review]

**Comment by tbjump** [comment]
@evan-gray wdyt? good to merge?

File: whitepapers/0009_guardian_key.md (hunk 1 above)
Comment (njkumr) [inline whitepapers/0009_guardian_key.md:30]: batch Vaa hash is for now calculated as `double-Keccak256(version | Keccak256(hash1 | hash2 | ... | hash_n))` where hash1 = double-Keccak256(observation1) and so on...

**Review by evan-gray — APPROVED** [review]
Concur with @tbjump that key usage may extend beyond node p2p specific code so documenting the expectations here seems appropriate.

**Review by mtjmp — APPROVED** [review]

**Pull request merged**